packages/solid/web/src/index.ts

```diff
@@ -99,9 +99,20 @@ export function Portal<T extends boolean = false, S extends boolean = false>(pro
   return marker;
 }
 
-type DynamicProps<T> = T & {
-  children?: any;
-  component?: Component<T> | string | keyof JSX.IntrinsicElements;
+type ValidElements = keyof JSX.IntrinsicElements;
+type ValidComponent<P> = (props: P) => JSX.Element;
```

Comment:
This is equivalent to [`Component`](https://github.com/solidjs/solid/blob/main/packages/solid/src/render/component.ts#L22), can we use that instead?

Reply:
Yes, I'll make that change.